### ISSUE TITLE
[SURE-7340] Fleet should never delete CRDs

### ISSUE THREAD

**Issue by manno** [rancher/fleet#1978]
When removing custom resource definitions, all custom resources are removed, too. This is unexpected and can lead to data loss during re-installation of charts. Helm also doesn't delete CRDs.
Fleet shouldn't delete CRDs which helm would not delete.

* safe default
* opt-in to delete them
* configurable via fleet.yaml (per bundle)

Why does fleet remove CRDs? Make sure there is no extra uninstall logic in Fleet. CRDs in template folder are not safe.

See:
* https://github.com/helm/community/blob/main/hips/hip-0011.md
* https://helm.sh/docs/howto/charts_tips_and_tricks/#tell-helm-not-to-uninstall-a-resource

**Comment by raulcabello** [comment]
# QA Template

## Solution
A new field called `deleteCRDResources` can now be added to the `fleet.yaml`. If  `deleteCRDResources` is set to true CRDs included in the deployment are deleted. Otherwise CRDs are kept in the cluster.


## Testing
- CRDs are deleted if `deleteCRDResources` is set to true
- CRDs are not deleted if `deleteCRDResources` is set to false or it is not present.

**Comment by sbulage** [comment]
---
**System Information** 
* Rancher version: 2.8.3-rc2
* Fleet Version: 0.9.1-rc.5

---
**Installation and Testing**

<details closed>
  <summary> <b>Rancher helm install command</b> </summary>

```
helm upgrade --install rancher rancher-latest/rancher  \
  --namespace cattle-system \
  --set hostname=<IP> \
  --set replicas=1 \
  --set bootstrapPassword=<PASSWD> \
  --set global.cattle.psp.enabled=false  \
  --version=v2.8.3-rc2 \
  --devel
```
</details>

**QA TEST PLAN**

Scenarios
|Scenario|Test Case|
|---|---|
|1| Test CRDs of deployed applications are removed when using `deleteCRDResources: true` in `fleet.yaml`|
|2| Test CRDs stays in the cluster of deployed applications when using `deleteCRDResources: false` in `fleet.yaml`|
|3| Test CRDs are not removed when upgrade Rancher from released version to 2.8.3-rc2 with `deleteCRDResources: false` in `fleet.yaml`|
|4| Test CRDs are removed with `deleteCRDResources: true` in `fleet.yaml` after upgrade Rancher from released version to 2.8.3-rc2 |
|5| Test CRDs are removed when upgrade Rancher from released version to 2.8.3-rc2 with `deleteCRDResources: true` in `fleet.yaml`|
---

**Comment by sbulage** [comment]
**TEST RESULT**

Scenarios
|Scenario|Test Case| Result
|---|---|---|
|1| Test CRDs of deployed applications are removed when using `deleteCRDResources: true` in `fleet.yaml`|:heavy_check_mark:|
|2| Test CRDs stays in the cluster of deployed applications when using `deleteCRDResources: false` in `fleet.yaml`|:heavy_check_mark:|
|3| Test CRDs are not removed when upgrade Rancher from released version to 2.8.3-rc2 with `deleteCRDResources: false` in `fleet.yaml`|
|4| Test CRDs are removed with `deleteCRDResources: true` in `fleet.yaml` after upgrade Rancher from released version to 2.8.3-rc2 |
|5| Test CRDs are removed when upgrade Rancher from released version to 2.8.3-rc2 with `deleteCRDResources: true` in `fleet.yaml`|
